[Documentation] UDF debugging instruction (#294)
dotnet/spark: docs/developer-guide.md

> Now you can run your .NET application with any debugger to debug your application.

Reviewer: What are the debuggers you are referring to? VS, I understand. Did we also verify VS Code C# debugging?

Author: This is only for VS. VS code can be done in a separate PR if needed.

Author: I will add VS info

### Debugging UDF

> **Note that this is currently supported only on Windows.**

Reviewer: For Linux, is [this](https://github.com/Microsoft/MIEngine/wiki/Offroad-Debugging-of-.NET-Core-on-Linux---OSX-from-Visual-Studio) going to be helpful?

Author: Probably not. `Debugger.Launch()` seems to work only for Windows: https://github.com/dotnet/corert/issues/7824

Author: That will work for driver side debugging.

Reviewer 2: I wrote "This _might_ not even work with CoreCLR" - I'm not a CoreCLR debugger dev. The issue was opened in the Core**RT** repo - if you need an authoritative answer from a CoreCLR debugger dev, you might want to open an issue in the CoreCLR repo.

Before running `spark-submit`, set the following environment variable:

> ```shell

Reviewer 3: ```suggestion
```bat
```

set DOTNET_WORKER_DEBUG=1
```
Now, when you run your Spark application, a `Choose Just-In-Time Debugger` window will pop up. Choose a debugger.

> The debugger will break at the following location in `TaskRunner.cs`:

Reviewer 3: ```suggestion
The debugger will break at the following location in [TaskRunner.cs](../src/csharp/Microsoft.Spark.Worker/TaskRunner.cs):
```

```C#
if (EnvironmentUtils.GetEnvironmentVariableAsBool("DOTNET_WORKER_DEBUG"))
{
    Debugger.Launch(); // <-- The debugger will break here.
}
```

> Now, navigate to the `cs` file that contains the UDF that you plan to debug, and set a breakpoint. (The breakpoint will say `The breakpoint will not currently be hit` because the worker hasn't loaded the assembly that contains UDF yet.)

Reviewer 3: ```suggestion
Now, navigate to the `.cs` file that contains the UDF that you plan to debug, and set a breakpoint. (The breakpoint will say `The breakpoint will not currently be hit` because the worker hasn't loaded the assembly that contains UDF yet.)
```

Hit `F5` to continue your application and the breakpoint will eventually be hit.

> **Note that the `Choose Just-In-Time Debugger` window will pop-up for each task. Therefore, make sure to set the number of exectuors to a low number. For example, you can use `--master local[1]` option for `spark-submit`.**

Reviewer: ```suggestion
**Note that the `Choose Just-In-Time Debugger` window will pop-up for each task. Therefore, make sure to set the number of executors to a low number. For example, you can use `--master local[1]` option for `spark-submit` to set the number of tasks to 1, and hence launching a single debugger instance.**
```

### Debugging Scala code

> If you need to debug the Scala side code (`DotnetRunner`, `DotnetBackendHandler`, etc.), you can use the following command, and attach a debugger to the running process using [Intellij](https://www.jetbrains.com/help/idea/attaching-to-local-process.html):

Reviewer: ```suggestion
If you need to debug the Scala side code (`DotnetRunner`, `DotnetBackendHandler`, etc.), you can use the following command, and attach a debugger to the running process using [IntelliJ](https://www.jetbrains.com/help/idea/attaching-to-local-process.html):
```
